config/config.go

```diff
@@ -18,12 +18,19 @@ func OptionalEnv(varName string, optional string) string {
 }
 
 /*
-	Check for an environment variable value, exit program if not found
+	Check for an environment variable value or the equivalent docker secret, exit program if not found
 */
 func RequiredEnv(varName string) string {
 	envVar := os.Getenv(varName)
-	if envVar == "" {
+	envVarFileName := os.Getenv(varName + "_FILE")
+	if envVar == "" && envVarFileName == "" {
 		log.Fatalf("The required env var %s is not provided. Exiting", varName)
+	} else if envVar == "" && envVarFileName != "" {
+		envVarFromFile, err := os.ReadFile(envVarFileName)
+		if err != nil {
+			log.Fatalf("Could not read env var from file %s. Exiting", envVarFileName)
```

Comment:
First, nice error message ! That would help the user.
Second, to go further, also send the "root" error back to the user. That would make debugging easier : was it a miss-configuration and file is not present? Or app can't read the file because of permission ? etc etc